lib/Sema/CSBindings.cpp

```diff
@@ -1629,11 +1667,30 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("delayed");
   if (isSubtypeOfExistentialType())
     attributes.push_back("subtype_of_existential");
-  auto literalKind = getLiteralKind();
+  auto literalKind = getLiteralForScore();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
-    attributes.push_back(literalAttrStr);
+    std::vector<std::string> kinds;
+    std::string literalAttrStr;
+    literalAttrStr.append("[literal: ");
+    if (literalKind == LiteralBindingKind::Atom) {
```

Comment:
Let's use a `switch` here instead of `if/if-else/else` to make sure that the checking is exhaustive so if anybody adds a new case to `LiteralBindingKind` they'd have to change this spot as well.